src/main/java/uk/gov/companieshouse/filinghistory/api/mapper/upsert/ResolutionChildMapper.java

```diff
@@ -0,0 +1,35 @@
+package uk.gov.companieshouse.filinghistory.api.mapper.upsert;
+
+import static uk.gov.companieshouse.filinghistory.api.mapper.DateUtils.stringToInstant;
+
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.filinghistory.InternalData;
+import uk.gov.companieshouse.api.filinghistory.InternalFilingHistoryApi;
+import uk.gov.companieshouse.api.filinghistory.Resolution;
+import uk.gov.companieshouse.filinghistory.api.model.mongo.FilingHistoryResolution;
+
+@Component
+public class ResolutionChildMapper implements ChildMapper<FilingHistoryResolution> {
+
+    private final DescriptionValuesMapper descriptionValuesMapper;
+
+    public ResolutionChildMapper(DescriptionValuesMapper descriptionValuesMapper) {
+        this.descriptionValuesMapper = descriptionValuesMapper;
+    }
+
+    @Override
+    public FilingHistoryResolution mapChild(FilingHistoryResolution resolution, InternalFilingHistoryApi request) {
+        InternalData internalData = request.getInternalData();
+        Resolution requestResolution = request.getExternalData().getResolutions().getFirst();
+
+        return resolution
+                .category(requestResolution.getCategory().getValue())
```

Comment:
Missing barcode and original_description

Reply:
need to double check but both seem to only get mapped at top-level in Perl

Reply:
mapped both in top level in this implementation, following Perl, can always change later if needed.